src/modules/preview-site/components/snapshot-row.tsx

```diff
@@ -0,0 +1,88 @@
+import { Spinner } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import { useI18n } from '@wordpress/react-i18n';
+import { useEffect } from 'react';
+import { ArrowIcon } from 'src/components/arrow-icon';
+import Button from 'src/components/button';
+import { useExpirationDate } from 'src/hooks/use-expiration-date';
+import { useFormatLocalizedTimestamps } from 'src/hooks/use-format-localized-timestamps';
+import { useSnapshots } from 'src/hooks/use-snapshots';
+import { useUpdateDemoSite } from 'src/hooks/use-update-demo-site';
+import { getIpcApi } from 'src/lib/get-ipc-api';
+import { PreviewActionButtonsMenu } from './preview-action-buttons-menu';
+import { ProgressRow } from './progress-row';
+
+interface SnapshotRowProps {
+	snapshot: Snapshot;
+	previousSnapshot: Snapshot | null;
+	selectedSite: SiteDetails;
+}
+
+export function SnapshotRow( { snapshot, previousSnapshot, selectedSite }: SnapshotRowProps ) {
```

Comment:
Let's rename the component to PreviewSiteRow

```suggestion
export function PreviewSiteRow( { snapshot, previousSnapshot, selectedSite }: SnapshotRowProps ) {
```

Reply:
Adjusted in https://github.com/Automattic/studio/pull/846/commits/58d02960f2e196678abf23bb5314359066b1e00e